Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -17,296 +17,526 @@
+}
 
-    delete _backing;
+static void __CGContextDeallocate(CFTypeRef cf) {
+    CGContextRef context = (CGContextRef)cf;
+    context->_impl.~__CGContextImpl();
 }
 
-CGContextImpl* __CGContext::Backing() {
-    return _backing;
+template <typename T, typename PT = typename std::remove_pointer<T>::type*, typename B = decltype(std::declval<PT>()->_base)>
+static PT _CFRuntimeCreateInstance(CFTypeID typeID, CFAllocatorRef allocator = kCFAllocatorDefault) {
+    return static_cast<PT>(
+        const_cast<void*>(_CFRuntimeCreateInstance(allocator, typeID, sizeof(typename std::remove_pointer<T>::type) - sizeof(B), nullptr)));
 }
 
 /**
  @Status Interoperable
 */
-void CGContextSetBlendMode(CGContextRef ctx, CGBlendMode mode) {
-    ctx->Backing()->CGContextSetBlendMode(mode);
+CFTypeID CGContextGetTypeID() {
+    static CFTypeID __kCGContextTypeID = _kCFRuntimeNotATypeID;
+    static const CFRuntimeClass __CGContextClass = { 0,    "CGContext", __CGContextInit,           NULL, __CGContextDeallocate, NULL,
+                                                     NULL, NULL,        __CGContextCopyDescription };
+
+    static dispatch_once_t initOnce = 0;
+    dispatch_once(&initOnce, ^{
+        __kCGContextTypeID = _CFRuntimeRegisterClass(&__CGContextClass);
+    });
+    return __kCGContextTypeID;
+}
+#pragma endregion
+
+static const wchar_t* TAG = L"CGContext";
+
+static void __CGContextInitializeWithRenderTarget(CGContextRef context, ID2D1RenderTarget* renderTarget) {
+    context->_impl.renderTarget = renderTarget;
+    CGContextSetRGBFillColor(context, 0, 0, 0, 0);
+    CGContextSetRGBStrokeColor(context, 0, 0, 0, 1);
 }
 
-CGBlendMode CGContextGetBlendMode(CGContextRef ctx) {
-    return ctx->Backing()->CGContextGetBlendMode();
+CGContextRef _CGContextCreateWithD2DRenderTarget(ID2D1RenderTarget* renderTarget) {
+    FAIL_FAST_HR_IF_NULL(E_INVALIDARG, renderTarget);
+    CGContextRef context = _CFRuntimeCreateInstance<CGContextRef>(CGContextGetTypeID());
+    __CGContextInitializeWithRenderTarget(context, renderTarget);
+    return context;
 }
 
+#pragma region Lifetime
 /**
  @Status Interoperable
 */
-void CGContextSetFillPattern(CGContextRef ctx, CGPatternRef pattern, const CGFloat* components) {
-    return ctx->Backing()->CGContextSetFillPattern(pattern, components);
+CGContextRef CGContextRetain(CGContextRef context) {
+    CFRetain((CFTypeRef)context);
+    return context;
 }
 
 /**
  @Status Interoperable
 */
-void CGContextSetPatternPhase(CGContextRef ctx, CGSize phase) {
-    return ctx->Backing()->CGContextSetPatternPhase(phase);
+void CGContextRelease(CGContextRef context) {
+    if (!context) {
+        return;
+    }
+
+    CFRelease((CFTypeRef)context);
 }
+#pragma endregion
 
+#pragma region Drawing Parameters
 /**
- @Status Stub
+ @Status Interoperable
 */
-void CGContextSetCharacterSpacing(CGContextRef ctx, CGFloat spacing) {
+void CGContextSetBlendMode(CGContextRef context, CGBlendMode mode) {
```

Comment:
I see your other comments on this. Its okay to be broken in terms of the annotations, if you can guarantee a scrub before we merge to develop (which is still some ways out).

---
In reply to: [82919343](https://github.com/Microsoft/WinObjC/pull/1126#discussion_r82919343) [](ancestors = 82919343)